### PR TITLE
Add map generation and display

### PR DESCRIPTION
**Pull request by gitesthuman**
# Pull request review checklist 📝 ✅

This is a checklist that helps conduct a pull request review properly. These points are required to merge the PR

- [x] There are no merge conflicts

- [x] The pull request is named accordingly (description of the functionality - it
   will be visible in the changelog later)
   
- [x] Every task requirement (from Jira) is done

- [ ] The app runs, it boots up properly (e.g. no syntax errors)

- [ ] The proposed functionality is working properly (you need to pull this branch
   and check for yourself)

- [ ] Each function, class or functionality is properly documented
   (short description of what it does, parameters or return values if they need
   any explanation)

### CODE REVIEW

**Comment by Silentsky0** [comment]
@gitesthuman trzeba dodać opencv-python do requirementsów

File: dnd-bot/dnd_bot/logic/prototype/entity.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ def __init__(self, x=0, y=0, sprite=None, name='Entity', id_game=0, skills=None)
         self.name = name
         self.id_game = id_game
         self.skills = skills
+        self.fragile = fragile
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/prototype/entity.py:15]: @gitesthuman czym jest fragile?? że może zostać zniszczone czy że może się przesuwać

File: dnd-bot/dnd_bot/logic/prototype/entity.py [hunk 2]
```diff
@@ -9,6 +12,10 @@ def __init__(self, x=0, y=0, sprite=None, name='Entity', id_game=0, skills=None)
         self.name = name
         self.id_game = id_game
         self.skills = skills
+        self.fragile = fragile  # bool: if entity can be moved or destroyed from its position
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/prototype/entity.py:15]: this description should be in `:param:`

File: dnd-bot/dnd_bot/logic/utils/utils.py [hunk 3]
```diff
@@ -11,3 +19,66 @@ def belongs_to_circle(x, y):
                 points.append((x, y))
 
     return points
+
+
+def paste_image(src: np.ndarray, dest: np.ndarray, x_offset: int, y_offset: int):
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:24]: not documented

File: dnd-bot/dnd_bot/logic/utils/utils.py [hunk 4]
```diff
@@ -11,3 +19,66 @@ def belongs_to_circle(x, y):
...
+    whole_map = cv.imread(game.sprite, cv.IMREAD_UNCHANGED)
+
+    # e1 = cv.getTickCount()
+    objects = [o for o in sum(game.entities, []) if o and not o.fragile]
+    for obj in objects:
+        paste_image(obj.sprite, whole_map, map_margin + obj.x * square_size, map_margin + obj.y * square_size)
+
+    # e2 = cv.getTickCount()
+    # t = (e2 - e1) / cv.getTickFrequency()
+    # print(f"game view processing time: {t} s")
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:48]: delete useless comments

File: dnd-bot/dnd_bot/logic/utils/utils.py [hunk 5]
```diff
@@ -11,3 +19,66 @@ def belongs_to_circle(x, y):
...
+
+    file_name = "dnd_bot/logic/utils/game_images/map%s.png" % game.token
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:50]: you could use an fstring here

File: dnd-bot/dnd_bot/logic/utils/utils.py [hunk 6]
```diff
@@ -11,3 +19,66 @@ def belongs_to_circle(x, y):
...
+                    map_margin + entity.y * square_size)
+
+    player_view = player_view[map_margin + (player.y - view_range) * square_size:
+                              map_margin + (player.y + view_range + 1) * square_size,
+                              map_margin + (player.x - view_range) * square_size:
+                              map_margin + (player.x + view_range + 1) * square_size, :]
+
+    # e2 = cv.getTickCount()
+    # t = (e2 - e1) / cv.getTickFrequency()
+    # print(f"player view processing time: {t} s")
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:77]: useless comments

File: dnd-bot/dnd_bot/logic/utils/utils.py [hunk 7]
```diff
@@ -11,3 +19,66 @@ def belongs_to_circle(x, y):
...
+
+    file_name = "dnd_bot/logic/utils/game_images/map%s.png" % game.token
+
+    cv.imwrite(file_name, whole_map)
+    del whole_map
+
+    return file_name
+
+
+def get_player_view(game: Game, player: Player):
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:58]: not documented

File: dnd-bot/dnd_bot/logic/utils/utils.py [hunk 8]
```diff
@@ -11,3 +19,66 @@ def belongs_to_circle(x, y):
...
+    player_view = player_view[map_margin + (player.y - view_range) * square_size:
+                              map_margin + (player.y + view_range + 1) * square_size,
+                              map_margin + (player.x - view_range) * square_size:
+                              map_margin + (player.x + view_range + 1) * square_size, :]
+
+    # e2 = cv.getTickCount()
+    # t = (e2 - e1) / cv.getTickFrequency()
+    # print(f"player view processing time: {t} s")
+
+    file_name = "dnd_bot/logic/utils/game_images/pov%s_%s.png" % (game.token, player.discord_identity)
```
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:79]: fstring here

**Review by Silentsky0 — CHANGES_REQUESTED** [review]

File: dnd-bot/dnd_bot/logic/utils/utils.py (hunk 4 above)
Comment (Silentsky0) [inline dnd-bot/dnd_bot/logic/utils/utils.py:48]: useless comments

Comment (gitesthuman) [reply dnd-bot/dnd_bot/logic/utils/utils.py:63]: needed for tests

**Review by Silentsky0 — APPROVED** [review]
LGTM

**Pull request merged**